array-sort.js

```diff
@@ -0,0 +1,59 @@
+var applyPatch = require("./apply-patch.js")
+var setNonEnumerable = require("./lib/set-non-enumerable.js")
+
+module.exports = sort
+
+function sort(compare) {
+    var obs = this
+    var list = obs._list.slice()
+
+    var unpacked = unpack(list)
+
+    var sorted = unpacked
+            .map(function(it) { return it.val })
+            .sort(compare)
+
+    var packed = repack(sorted, unpacked)
+
+    //fake diff - for perf
+    //adiff on 10k items === ~3200ms
+    //fake on 10k items === ~110ms
+    var changes = [ [ 0, packed.length ].concat(packed) ]
```

Comment:
Having the diff say "oh btw, everything was removed and everything was added" seems sub optimum.

The real challenge of this implementation was to do a complete immutable implementation of `sort()` in userland that build up the diff as it went along.